tree/asset_tree_cache.go

```diff
@@ -0,0 +1,86 @@
+package tree
+
+import (
+	"sync"
+
+	bsmt "github.com/bnb-chain/zkbnb-smt"
+	lru "github.com/hashicorp/golang-lru"
+)
+
+// Lazy init cache for asset trees
+type AssetTreeCache struct {
+	initFunction      func(index int64) bsmt.SparseMerkleTree
+	nextAccountNumber int64
+	mainLock          sync.RWMutex
+	changes           map[int64]bool
+	changesLock       sync.RWMutex
+	treeCache         *lru.Cache
+}
+
+// Creates new AssetTreeCache
+// maxSize defines the maximum size of currently initialized trees
+// accountNumber defines the number of accounts to create/or next index for new account
+func NewLazyTreeCache(maxSize int, accountNumber int64, f func(index int64) bsmt.SparseMerkleTree) *AssetTreeCache {
+	cache := AssetTreeCache{initFunction: f, nextAccountNumber: accountNumber, changes: make(map[int64]bool, maxSize*10)}
+	cache.treeCache, _ = lru.NewWithEvict(maxSize, cache.onDelete)
+	return &cache
+}
+
+// Updates current cache with new(updated) init function and with latest account index
+func (c *AssetTreeCache) UpdateCache(accountNumber int64, f func(index int64) bsmt.SparseMerkleTree) {
```

Comment:
The `initFunction` field is only assigned in the function `UpdateCache`, and `UpdateCache` is only called in `RegisterZnsExecutor` and `WitnessHelper`. 
It means for committer,  the `initFunction` is only assigned after the executing a `RegisterZns` transaction, otherwise it will be nil.

Reply:
And `tree.NewEmptyAccountAssetTreeFunc(bc.StateDB().TreeCtx, uint64(bc.CurrentBlock().BlockHeight))` will put `BlockHeight` as an env in this function, which may not correct as  the cache processing new blocks,